Tools/llm-tool/README.md

```diff
@@ -60,3 +60,187 @@ Building in Release / optimizations will remove a lot of tail calls in the C++
 layer.  These lead to the stack overflows.
 
 See discussion here: https://github.com/ml-explore/mlx-swift-examples/issues/3
+
+## LoRA
+
+`llm-tool` provides an example LoRA driver based on:
+
+- https://github.com/ml-explore/mlx-examples/blob/main/lora/README.md
+
+This is an example of using MLX to fine-tune an LLM with low rank adaptation
+(LoRA) for a target task.[^lora] The example also supports quantized LoRA
+(QLoRA).[^qlora] The example works with Llama and Mistral style models
+available on Hugging Face.
+
+In this example we'll use the WikiSQL[^wikisql] dataset to train the LLM to
+generate SQL queries from natural language. However, the example is intended to
+be general should you wish to use a custom dataset.
+
+> Note: Some of the prompts have newlines in them which is difficult to achieve via running in Xcode.
```

Comment:
Can you explain that comment?

Reply:
Sure, for testing this you need to pass a prompt like this:

```
    --prompt "table: 1-10015132-16
columns: Player, No., Nationality, Position, Years in Toronto, School/Club Team
Q: What is terrence ross' nationality
A: "
```

The newlines seem to be important.  You can put that in the Xcode parameters field but it seems to eat the newlines:

<img width="694" alt="image" src="https://github.com/ml-explore/mlx-swift-examples/assets/46639364/6178ecbf-9cae-4128-bbfc-8379751e39ed">

Reply:
I see, that makes sense.